012_super_resolution/STORM.ipynb

```diff
@@ -0,0 +1,818 @@
OJ9222aHYtPklEoFSZLPoV7oGSCfEyNjguepUrYV/ffVY7LXUUu/WkTTGj12mwSXw4bZlYXQQ1LpKHrhe+ac7/Xg+QOtimEnGjpmfYRX2VsxrSzBHmLYJFnc3H6mF+2DI+h0+/DW6W7D8u++YBj/NEgMrdc2bUU7zuBIENvO9Cjhn88faFXlTOOIiY45Rj0nFIko1fneOt2NEx3ytoNhFtcfJQmYXVWIFdNKDbfPfxMKR9AYTaIfJ9waCmT6WNWdXzka7wEHBlUy9H4DQ10J34seQM+wH409nlj4nl3CtQuqsXpWueo34v3CRd3ReD+lMwwpnV5YeuGyemPF5pNdupXx9A5L+5GeJ6jDJgm50GL70xMytO3ccqo7zj6abiJC6J17UdQSnztm8NxpyXKyQxabRUcp7Zg/7A/h+QOt6v1J6r8Zk/M1bTnVbXivFTjtlp9VVkKlATl8T0wqL27f7OVRmEWfO8Iq3kDYcri3EfNqChOKwpUF5l6xWkiUIgiB+rJ8PHrbCkWk+tkbDfjY/+5U3nSlUqGGIAiCIMaKQa8c0iXO5fWmmjkmuXwAwTsnuqGggWFihp4IwMPnZ1fKb9H9oQiGfCG0DYzArklabJMkbD4lT/gZS676Hq/upoeVsLVk8trowT1WjNp4tG3QsBrfzsbeOGFKiy8YNnw7X57vRDDMEoYgMiQ+D3pCzNluj9IGQP2m3WxeVJybg1tXTE0qNNIWNZS7hnzY1din6xUmYpaHRS9czYr4aJaonBu3vmBYEQhEDxm989ul8QiMMIZejeeIWbNigrG88X4hHxz/WSjMMOwPCnmuYsj3kiSESFkTFbVoV42w2O+H/SE09RqEpwmeXk29Xrzd0INDLQNK35EAlOTlqO5BHqan9aCJ/RsrhFBflpxHRjpIpzWgF+qs9SI1Q8+DVNuf9O5Bjz+Efm8AgZD6+WHUJ8KCgKU3FmnvrWTOEb/OicysYJhpwliTuxJiG7XtTTakzojZVYWWvZGCJoUZRLqGfIYFB3i4sB6bjnfirdPq/IYRFhPFrQhTeqtYeV4m+0glUYogTPj2rYvx1L+uUlyP3//zbXh04ynl+3RVWCEIgiCIVNja0I1djX26XlFWDA2+Xpw3RnSybDnkhQHH2+OrR9kk2TuFV/XjYWbnej1K6XZOQ5ecY0T2lNLJKWWye1XJcQ0jwXBC49us1LwV+Dk301/ME9nqvwXniCKG9vvuYT8ijFmqTKZ3HkTDTkzIG9thdL3ooniujHKHpYroSSfu0wjTsFTNr+02Cae7jCuccYYthEK+crQDh6OeIqKAa8UT7UTHEM5rKjyahabyPsXt121nYkYmP/5+bwB+QVzQOy96BnwqSbc5Q74g2oXcY9qwWED2nonluGPY39yvfKet3ia22UjI1AuRm19TqIRliqiE+nFK1pwukrEvLOX80TmfI1GRuSc6fnD8Brl1IxqBML4h8j9XzK1EcW6Oecgy4oVTQBZfzCpjnuhwo1XwHBVFcSPRRkQMMdZW6rRyL1i5W5IR4ZO5/TYKqWXK8mMvYbqGfHD7giqvK7G/93sDceO0XN0WeM/yKbg46slq3Jb449Eeo9YzORVIlCIIE4pyc7Aqmvw1FI7gslnlinvqSCCMJQ+9gl9vO6cs/37neaWEKkEQBEGMB/3egGZCKaHT7VNNRE1FKcGVn0/srZaeFhkciX97HmFMtQ3+V/eQXwlB4qg9TuLDC8zyPpkZJO2DI9h9zvhtMiB7/6wVKp2lSjoSvuoaATrnkL8wU9axkJtH7yxpKwWKdA354vJHJfK6Gw02SYJoj2u7bTLFaLS/tert7tPJ22OGmK+He5WZoevxZtJt+HEcahmIC/fRXvLzfZ64NvHNx0KkYp+LeX0SoT17x9rdhtUutfvW7heQxRAg1j/FW4fbuLFk9fIfYm4rvn5Rbg7WXVSj6v9DGiM8URJrK6jCYMcyizp4qFjq+4gbOxNUgjTzjFWeDdFVjJ4l/BRXFrpw3cJq1fXQO5Z+bwB7z/crbeBsPW08HgHAwZYB5W8xWboV8pz2uM942Khfp8pkqnkM9TjaNhj3WaqifoHLgUumx8Ki3zjRhYMtA2gfHMGLh9oStjoSifVnrZe0Ft3wUAvPZvKUIogxwmG34YFbLsIHV9YDkB+Mn712DpbVlwAAznQP4+vPHsGx6AP6cMsgrvzP15UEpW0DI3jhYFva3yoSBEEQFzbhiFqSYYxhx9lelRFuJkrZJClm3GoMD6tvfbuH/bql48MR9eTUqmij5yllRp5GoElFGzLKB2UFvj+jpNupbEukR/Aq499fu6BKVV3YkUAsOto6qIg6Zp5lImKiY62hPxbYJEmVP0orQh3RGHZWLrMVDwrV+kleQofgJXCmO95INtteca7c58yOg9+LI8Gw4p3F0RrN3BNEvG/tNkmVIHo8E1Ew6CdqF5E0/wJCBVHNuk67zVBcVHmGRoCAoG6mQyx2Oey4PponTQ6FlXdYV6Jf3GC0WBUMrRya2Sr9ngB2nDVOaM67kpI/zHA94704DDxpeGVH8dolShgvYsXzUUSvEAX3sgsmEK3Nkqynwu5zfdh3fiBt2+vzBLCrsQ/hSOJ8kIFwTHivKJS9rowSk5v1nUuml2FWZXxxBiB570QSpQgiRcoKnLh3/XxcHFWqF9UVY+fX1uHKuXLpW6fDhtUzy1EVdV/dcbYX//7H/egblicMrx7twN2/3YvB6BtIvcomBEEQBGGFc4LXkV4YTZ/HWFAoyc9RjNtwtIoai9pzyVSl0ws5CUfDBDhW56lDvlBSwhI37pPFqKx8sihhVAYV6KwyOBLUDV9Rix3yvlwOO4rzYmFL2hxdWjrcPrwd9YqqKHQq8xU99ML4Yl4NYzdX0Rq2WqOzZ9i6Z8/RtsGUqiknMui0c7VkvXBE8ZB7bpiHIcY4pwn7M7IYxfOYTDiRGVpBzAo8n5VeVUUteuF72pxSHW4fXoomtjc7qs2nurD5ZKzSWbqi
```

Comment:
In the previous figure the non-selected pixels are set to zero.  Here they are sampled from Laplace distribution. I think this is inconsistent and should be aligned if possible.  I didn't understand the motivation for sampling the non-selected pixels through Laplace, when we consider them known to have value zero, since they do not contribute through the forward model.

Reply:
Yes, it is correct that $x_{\mathcal{I}^c}$ doesn't contribute to the likelihood computation, but they still follow a Laplace distribution. To put it another way, we update our belief of $x_{\mathcal{I}}$ by combining a Laplace and an approximated likelihood, but we leave $x_{\mathcal{I}}^c$ to be the same as our prior believe, i.e., Laplace distribution.
Thus in the previous plot on the posterior mean: 
- for $x_{\mathcal{I}}$, the posterior mean is computed from our NUTS samples, 
- for $x_{\mathcal{I}^c}$, the posterior mean is the same as the prior mean, i.e., zeros.

And here to draw one sample/realization of the whole image:
- for $x_{\mathcal{I}}$, a sample is directly from the NUTS sampling,
- for $x_{\mathcal{I}^c}$, the posterior is the the same as the prior, so we draw from the Laplace.

Reply:
The above is the procedure described in the paper, so we decided to stick to it. We added a comment in the notebook to make it clear that:
> More specifically, for $x_{\mathcal{I}}$, the posterior mean is computed from our NUTS samples, while for $x_{\mathcal{I}^c}$, the posterior mean is the same as the prior mean, i.e., zeros.